src/Config/Processor/BuilderProcessor.php

```diff
@@ -126,30 +138,53 @@ private static function processFieldBuilders(array $fields)
             }
 
             if ($fieldBuilderName) {
-                $buildField = self::getBuilder($fieldBuilderName, self::BUILDER_FIELD_TYPE)->toMappingDefinition($builderConfig);
-                $field = \is_array($field) ? \array_merge($buildField, $field) : $buildField;
+                $mapping = self::getBuilder($fieldBuilderName, self::BUILDER_FIELD_TYPE)->toMappingDefinition($builderConfig);
+
+                $fieldMapping = $mapping['field'] ?? $mapping;
+                $field = \is_array($field) ? \array_merge($fieldMapping, $field) : $fieldMapping;
+
+                if (isset($mapping['field'], $mapping['types'])) {
+                    $types = \array_merge($types, $mapping['types']);
```

Comment:
Check added types in '$mapping['types']' not in `$types` already. Throw if so, including type name and builder that emitteed the duplicate.